core/corehttp/gateway_test.go

```diff
@@ -20,18 +21,33 @@ import (
 	testutil "github.com/ipfs/go-ipfs/util/testutil"
 )
 
-type mockNamesys map[string]path.Path
+type mockNamesys map[string]mockEntry
 
-func (m mockNamesys) Resolve(ctx context.Context, name string) (value path.Path, err error) {
-	return m.ResolveN(ctx, name, namesys.DefaultDepthLimit)
+type mockEntry struct {
+	path path.Path
+	ttl  time.Duration
 }
 
-func (m mockNamesys) ResolveN(ctx context.Context, name string, depth int) (value path.Path, err error) {
-	p, ok := m[name]
+func (m mockNamesys) Resolve(ctx context.Context, name string) (path.Path, error) {
+	p, _, err := m.ResolveDetailed(ctx, name)
+	return p, err
+}
+
+func (m mockNamesys) ResolveN(ctx context.Context, name string, depth int) (path.Path, error) {
+	p, _, err := m.ResolveNDetailed(ctx, name, depth)
+	return p, err
+}
+
+func (m mockNamesys) ResolveDetailed(ctx context.Context, name string) (path.Path, time.Duration, error) {
```

Comment:
`ResolveWithTTL` sounds good. `ResolveWithCaching` might be ambiguous since this variant of `Resolve` does not affect whether caching occurs at all.